src/server/pfs/server/api_server.go

```diff
@@ -428,6 +428,15 @@ func (a *apiServer) putFileObj(ctx context.Context, objClient obj.Client, reques
 				defer func() {
 					<-sem
 				}()
+
+				if strings.HasSuffix(name, "/") {
+					// Amazon S3 supports objs w keys that end in a '/'
+					// PFS needs to treat such a key as a directory.
+					// In this case, we rely on the driver PutFile to
+					// construct the 'directory' diffs from the file prefix
```

Comment:
Hmm, is it? This is the issue sean saw where a user had data in s3 that included a file with a `/` suffix from a weird sync to s3 and it caused conflicts.